### PR TITLE
chore: use docker hub image

### PR DESCRIPTION
**Pull request by kenhuang**
Description:

* upgrade SONAR_SCANNER_VERSION to 4.3
* remove dependency on `binaries.sonarsource.com`

### CODE REVIEW

**Comment by simonbrandhof** [comment]
Nice contribution, thanks!

**Comment by simonbrandhof** [comment]
Commit message was reworded. It's now merged into `master`. Closing https://jira.sonarsource.com/browse/SCCOMM-9.

**Comment by simonbrandhof** [comment]
My bad, https://jira.sonarsource.com/browse/SCCOMM-9 is another optimization.

File: Dockerfile [hunk 1]
```diff
@@ -12,29 +12,12 @@ LABEL "com.github.actions.color"="green"
...
-    && rm sonar-scanner-cli.zip \
-    && mv sonar-scanner-${SONAR_SCANNER_VERSION} ${SONAR_SCANNER_HOME}
-
-RUN wget -q -O nodejs.tar.xz https://nodejs.org/dist/${NODEJS_VERSION}/node-${NODEJS_VERSION}-linux-x64.tar.xz \
-    && tar -Jxf nodejs.tar.xz \
-    && mv node-${NODEJS_VERSION}-linux-x64 ${NODEJS_HOME}
-
-RUN npm install -g typescript
+# https://help.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions#user
+USER root
```
Comment (janos-ss) [inline Dockerfile:20]: Hi @kenhuang, I have a question about this change here.

The linked page says to NOT use the `USER` instruction... I see that the default is supposed to be `root` anyway, but if that's the default, then why use this at all?

This question came up through an [issue reported in our community](https://community.sonarsource.com/t/github-actions-sonarcloud-github-action-permissions-issue/41609), which to me seems related to this code (I could be wrong).

If you have some insight, I would appreciate it, thanks!

Comment (kenhuang) [reply Dockerfile:20]: Think before [this](https://github.com/SonarSource/sonar-scanner-cli-docker/commit/51d988004ef01eba4300e16d2b809d354daff087#diff-be21fb71aa777f5402266b039ef614ceec518cfa83e698ccc73d60ec14e04852L78) change upstream docker image was using scanner-cli, this line from what I remember was to set it back to root. 

The issue you mentioned above seems related to this GitHub self hosted runner issue: https://github.com/actions/runner/issues/434

**Pull request closed without merging**